From d7d0910a208ea0071d2c4ecae8ddde03651b9a39 Mon Sep 17 00:00:00 2001
From: Terminator-J <terminator_j@hotbutteredmonkey.com>
Date: Mon, 21 Jul 2025 00:16:56 -0700
Subject: [PATCH] Fixup! "erofs: Adapt 4.9 getattr inode_operations"

This partially reverts commit 962b5bd10873c2324ae71ac8cb91e5d3bf6c5ad1.

Reason for revert:
Not needed any longer after "statx: Add a system call to make enhanced file info available" inclusion, fixes build breakage.

Still skip updating attributes_mask as it's not implemented currently.
---
 fs/erofs/inode.c    | 8 ++++----
 fs/erofs/internal.h | 4 ++--
 2 files changed, 6 insertions(+), 6 deletions(-)

diff --git a/fs/erofs/inode.c b/fs/erofs/inode.c
index c7fd150add27..a04a9d3772ea 100644
--- a/fs/erofs/inode.c
+++ b/fs/erofs/inode.c
@@ -337,16 +337,16 @@ struct inode *erofs_iget(struct super_block *sb,
 	return inode;
 }
 
-int erofs_getattr(struct vfsmount *mnt, struct dentry *dentry,
-			struct kstat *stat)
+int erofs_getattr(const struct path *path, struct kstat *stat,
+		  u32 request_mask, unsigned int query_flags)
 {
-	struct inode *const inode = d_inode(dentry);
+	struct inode *const inode = d_inode(path->dentry);
 
-#if 0
 	if (erofs_inode_is_data_compressed(EROFS_I(inode)->datalayout))
 		stat->attributes |= STATX_ATTR_COMPRESSED;
 
 	stat->attributes |= STATX_ATTR_IMMUTABLE;
+#if 0
 	stat->attributes_mask |= (STATX_ATTR_COMPRESSED |
 				  STATX_ATTR_IMMUTABLE);
 #endif
diff --git a/fs/erofs/internal.h b/fs/erofs/internal.h
index c52e77109675..7369d1fe304f 100644
--- a/fs/erofs/internal.h
+++ b/fs/erofs/internal.h
@@ -382,8 +382,8 @@ extern const struct inode_operations erofs_symlink_iops;
 extern const struct inode_operations erofs_fast_symlink_iops;
 
 struct inode *erofs_iget(struct super_block *sb, erofs_nid_t nid, bool dir);
-int erofs_getattr(struct vfsmount *mnt, struct dentry *dentry,
-			struct kstat *stat);
+int erofs_getattr(const struct path *path, struct kstat *stat,
+		  u32 request_mask, unsigned int query_flags);
 
 /* namei.c */
 extern const struct inode_operations erofs_dir_iops;
